src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -109,19 +112,24 @@ class ResourceHashtableBase : public STORAGE {
   int number_of_entries() const { return _number_of_entries; }
 
   bool contains(K const& key) const {
-    return get(key) != NULL;
+    return const_cast<ResourceHashtableBase*>(this)->get(key) != NULL;
```

Comment:
Wonder if we can macro out the pattern of un-constifying "this". I would hope that C++ develops and gives us a better looking solution.